lib/ansible/modules/cloud/vmware/vmware_guest_facts.py

```diff
@@ -121,10 +131,7 @@ def getvm(self, name=None, uuid=None, folder=None):
         if uuid:
             vm = find_vm_by_id(self.content, vm_id=uuid, vm_id_type="uuid")
         elif folder:
-            # Build the absolute folder path to pass into the search method
-            if not self.params['folder'].startswith('/'):
-                self.module.fail_json(msg="Folder %(folder)s needs to be an absolute path, starting with '/'." % self.params)
-            searchpath = '%(datacenter)s%(folder)s' % self.params
+            searchpath = self.params['folder']
```

Comment:
Could this cause backwards compatibility issues?  If someone was relying on the previous behavior of adding the datacenter prefix to searchpath?

I think the approach in this pr is better and more explicit. But I'm not sure if the change will introduce api differences from the 2.3 version though. Would users see a behavior change?